scipy/signal/_signaltools.py

```diff
@@ -4458,32 +4458,32 @@ def decimate(x, q, n=None, ftype='iir', axis=-1, zero_phase=True):
     if n is not None:
         n = operator.index(n)
 
+    result_type = x.dtype
```

Comment:
you can use `if x.dtype is not np.inexact:` test here which also covers `np.float16` case.

Reply:
OK - I didn't actually change this behaviour, just moved it earlier (needed for different FIR & IIR handling).  It seems like a reasonable change.

Reply:
@roryyorke want to try this change and add a `float16` case in the tests somewhere? Then this LGTM +1 for merge!